scripts/windows/test/Run-E2ETest.ps1

```diff
@@ -1087,18 +1087,26 @@ Function RunTransparentGatewayTest
     $testCommand = AppendInstallationOption($testCommand)
     Invoke-Expression $testCommand | Out-Host
 
+    $testExitCode = $LastExitCode
```

Comment:
This test command is used for deploying edge runtime and modules.  If this command failed, we should fail this test immediately.  Also I can see $testExitCode is only assigned here and return at the end of this function.  Therefore failure of any LeafDevice Test will not affect to return value of this test. Please make sure any leaf device test failure should fail this test as well.

Reply:
I can return immediately when deployment fails. It may not even be necessary to add it to $testExitCode so we don't contaminate it with non-test results.

All the LeafDevice tests use $testExitCode so if any of the tests fail, they append 1 to the result. The command that Damon proposed will parse the collection of results and return 1 if any of them failed, so I think this is handled here.

Reply:
I think we should avoid to use same variable name in 2 different methods - RunTransparentGatewayTest and RunLeafDeviceTest.  That's why an array of test result code from RunTransparentGatewayTest method.  And this is not easy for other devs to follow up.  We should capture result code explicitly from each RunLeafDeviceTest calls in RunTransparentGatewayTest method.  And just return return a non-zero value from RunTransparentGatewayTest method if any of these tests are failed.  @damonbarry what do you think?

Reply:
My suggestion was simply less code, and less clutter. You can do it either way, though. This script will being going away once the new end-to-end tests are in place anyway.

Reply:
I didn't know that the script will be going away. If that's the case, I don't want to change things too much. I'll leave the changes as is, with LeafDevice tests using $testExitCode.